packages/wpcom-proxy-request/src/index.js

```diff
@@ -18,6 +18,8 @@ const proxyOrigin = 'https://public-api.wordpress.com';
  */
 const origin = window.location.protocol + '//' + window.location.host;
 
+let onStreamRecord = null;
```

Comment:
This is a global variable, but it would be shared by all requests running in parallel. It's better to store the callback as `xhr.onStreamRecord = ...`, then store the `xhr` in `requests[ id ]`, and then retrieve it before called:
```js
if ( statusCode === 207 ) {
  const onStreamRecord = xhr.onStreamRecord;
  if ( onStreamRecord ) {
    onStreamRecord( body );
  }
}
```